spec/gutter-spec.coffee

```diff
@@ -0,0 +1,74 @@
+{Gutter, DEFAULT_PRIORITY} = require '../src/gutter'
+
+describe 'Gutter initialization tests.', ->
+  name = 'name'
+  fakeGutterContainer = {}
+
+  it 'Test initializing a Gutter with all options.', ->
+    location = 'left'
+    priority = 1
+    visibleInitially = false
+    options =
+      'location': 'left'
+      'priority': 1
+      'visibleInitially': false
+    gutter = new Gutter name, fakeGutterContainer, options
+    expect(gutter.name).toBe name
+    expect(gutter.gutterContainer).toBe fakeGutterContainer
+    expect(gutter.location).toBe location
+    expect(gutter.priority).toBe priority
+    expect(gutter.visible).toBe visibleInitially
+
+  it 'Test initializeing a Gutter with minimal options.', ->
+    fakeGutterContainer = {};
+    gutter = new Gutter name, fakeGutterContainer
+    expect(gutter.name).toBe name
+    expect(gutter.gutterContainer).toBe fakeGutterContainer
+    expect(gutter.location).toBe 'left' # default
+    expect(gutter.priority).toBe DEFAULT_PRIORITY
+    expect(gutter.visible).toBe true # default
+
+describe 'Test hiding and showing a gutter.', ->
+  fakeGutterContainer = {}
+  fakeListener =
+    onGutterShow: null
+    onGutterHide: null
+  name = 'name'
+
+  attachListenerToGutter = (gutter) ->
+    gutter.onShow () ->
+      fakeListener.onGutterShow()
+    gutter.onHide () ->
+      fakeListener.onGutterHide()
+    spyOn fakeListener, 'onGutterShow'
+    spyOn fakeListener, 'onGutterHide'
+
+  it 'Test hiding a visible Gutter.', ->
+    options = visibleInitially: true
+    gutter = new Gutter name, fakeGutterContainer, options
+    attachListenerToGutter(gutter)
+
+    expect(gutter.visible).toBe true
+    gutter.hide()
+    expect(gutter.visible).toBe false
+    expect(fakeListener.onGutterHide).toHaveBeenCalled()
+
+  it 'Test showing a hidden Gutter.', ->
+    options = visibleInitially: false
+    gutter = new Gutter name, fakeGutterContainer, options
+    attachListenerToGutter(gutter)
+
+    expect(gutter.visible).toBe false
+    gutter.show()
+    expect(gutter.visible).toBe true
+    expect(fakeListener.onGutterShow).toHaveBeenCalled()
+
+
+describe 'Test destroying a Gutter.', ->
+  fakeGutterContainer = removeGutter: null
+
+  it 'Test destroying a Gutter notifies its Container.', ->
+    gutter = new Gutter 'name', fakeGutterContainer
+    spyOn fakeGutterContainer, 'removeGutter'
```

Comment:
You could build the `gutterContainer` on the fly with an anonymous spy in this case.

```coffee
gutterContainer = removeGutter: jasmine.createSpy("removeGutter")
```